### PR TITLE
Implemented anti-spam fee query

### PR DESCRIPTION
**Pull request by ruseinov**
Fixes https://github.com/iov-one/weave/issues/265 leftovers

### CODE REVIEW

**Comment by ruseinov** [comment]
@ethanfrey this is ready for review

File: cmd/bnsd/app/app.go [hunk 1]
```diff
@@ -122,16 +125,16 @@ func Stack(issuer weave.Address, nftBuckets map[string]orm.Bucket, minFee coin.C
 // the given arguments. If you are not sure what to use
 // for the Handler, just use Stack().
 func Application(name string, h weave.Handler,
-	tx weave.TxDecoder, dbPath string, debug bool) (app.BaseApp, error) {
+	tx weave.TxDecoder, dbPath string, options *server.Options) (app.BaseApp, error) {
```
Comment (ethanfrey) [inline cmd/bnsd/app/app.go]: Yeah this is useful to have all args here.

File: cmd/bnsd/app/app.go [hunk 2]
```diff
@@ -101,6 +103,7 @@ func QueryRouter() weave.QueryRouter {
 		orm.RegisterQuery,
 		currency.RegisterQuery,
 		distribution.RegisterQuery,
+		antiSpamQuery.RegisterQuery,
```
Comment (ethanfrey) [inline cmd/bnsd/app/app.go]: Interesting.. I would assume that antiSpamQuery could be a RegisterQuery function already

File: x/msgfee/antispam_query.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// AntiSpamQuery allows querying currently set anti-spam fee
+type AntiSpamQuery struct {
+	minFee coin.Coin
+}
+
+func NewAntiSpamQuery(minFee coin.Coin) *AntiSpamQuery {
+	return &AntiSpamQuery{minFee: minFee}
+}
+
+func (q *AntiSpamQuery) Query(db weave.ReadOnlyKVStore, mod string, data []byte) ([]weave.Model, error) {
```
Comment (ethanfrey) [inline x/msgfee/antispam_query.go]: Okay, so it ignors mod and data, and anything returns the same value.
Could enforce `mod == "" && len(data) == 0` but I guess this is more user-friendly.

Nice implementation by the same. Simple and isolated from other code.

File: x/msgfee/antispam_query_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+func TestAntiSpamQuery(t *testing.T) {
+	initialCoin := coin.Coin{
+		Whole:      0,
+		Fractional: 0,
+		Ticker:     "DOGE",
+	}
+
+	q := NewAntiSpamQuery(initialCoin)
+	model, err := q.Query(nil, "", nil)
+	if err != nil {
```
Comment (ethanfrey) [inline x/msgfee/antispam_query_test.go]: You can use `assert.NoNil(err)` for this same effect (generally for all the fatal nil or error checks, easier to use it)
Import from weavetest/assert

File: x/msgfee/antispam_query_test.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		t.Fatalf("expected 1 model, got %d", modelNum)
+	}
+
+	c := coin.Coin{}
+	err = c.Unmarshal(model[0].Value)
+	if err != nil {
+		t.Fatalf("unexpected unmarshaling error: %s ", err.Error())
+	}
+
+	if !(c).Equals(initialCoin) {
```
Comment (ethanfrey) [inline x/msgfee/antispam_query_test.go]: why `(c).` not `c.` ?

File: x/msgfee/antispam_query_test.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"testing"
+
+	"github.com/iov-one/weave/coin"
+)
+
+func TestAntiSpamQuery(t *testing.T) {
+	initialCoin := coin.Coin{
+		Whole:      0,
```
Comment (ethanfrey) [inline x/msgfee/antispam_query_test.go]: Please have non-zero values in the test case

File: cmd/bnsd/app/app.go (hunk 2 above)
Comment (ethanfrey) [reply cmd/bnsd/app/app.go]: After reading the code, very nice choice :+1:

**Comment by ruseinov** [comment]
I've fixed the comments and merged this. In terms of ATs - I'm not sure it's worth the effort, we have this unit-tested, the code is super simple too. 

I'm sure though when implementing a dynamic minFee update we will add ATs as the logic will become a bit less complex.

**Pull request merged**